### PR TITLE
compatibility with Twig 2.0

### PR DESCRIPTION
**Pull request by OskarStark**
I am targetting this branch, because this is BC.

Refs #1183 

## Changelog

```markdown
### Added
- support for Twig 2.0
```

## Subject

I was wondering, as there wasn't any dependency for twig...

### CODE REVIEW

**Comment by greg0ire** [comment]
No code changes besides `composer.json` ?

**Comment by OskarStark** [comment]
> No code changes besides composer.json ?

I think so, but lets run travis once

**Comment by enekochan** [comment]
As for what I've seen in [Twig deprecations from 1.x to 2.x](http://twig.sensiolabs.org/doc/1.x/deprecated.html#extensions) `Twig_ExtensionInterface::getName()` has to be removed (`SonataFormatterBundle/Extension/BaseProxyExtension.php` implements `Twig_ExtensionInterface`):

https://github.com/sonata-project/SonataMediaBundle/blob/3.x/Twig/Extension/FormatterMediaExtension.php#L81

https://github.com/sonata-project/SonataMediaBundle/blob/3.x/Twig/Extension/MediaExtension.php#L76

**Comment by OskarStark** [comment]
> SonataFormatterBundle/Extension/BaseProxyExtension.php implements Twig_ExtensionInterface

but this is the formatter bundle, not the media bundle

**Comment by enekochan** [comment]
The [commit that introduced the deprecation in Twig 1.26](https://github.com/twigphp/Twig/commit/f58cce77d773a805e5e4d0ddf5b5e5b17b097d93) says that Twig extensions should inherit from ``Twig_extension`` and not implement ``Twig_ExtensionInterface`` as SonataFormatterBundle/Extension/BaseProxyExtension does.

Would this mean that SonataFormatterBundle and all the bundles that extend ``BaseProxyExtension``from it to create Twig extension should be changed?

**Comment by enekochan** [comment]
There are some more changes needed to achieve compatibility with Twig 2. I don't know if I should open a new PR so I'll explain what I've done in case @OskarStark wants to change his PR.

For both `Sonata\MediaBundle\Twig\Extension\FormatterMediaExtension.php` and `Sonata\MediaBundle\Twig\Extension\MediaExtension.php`

* Remove `getName` function.
* Change in `getTokenParsers` function the reference of `getName` to `self::class`. So change this:

```php
    public function getTokenParsers()
    {
        return array(
            new MediaTokenParser($this->getName()),
            new ThumbnailTokenParser($this->getName()),
            new PathTokenParser($this->getName()),
        );
    }
```

To this:

```php
    public function getTokenParsers()
    {
        return array(
            new MediaTokenParser(self::class),
            new ThumbnailTokenParser(self::class),
            new PathTokenParser(self::class),
        );
    }
```

With those changes my app worked fine both with Twig 1.31.0 and Twig 2.1.0

**Comment by greg0ire** [comment]
> I don't know if I should open a new PR so I'll explain what I've done in case @OskarStark wants to change his PR.

You can always make a PR on @OskarStark branch ;)

**Comment by enekochan** [comment]
Done :) Hope everything is all right. Ping @OskarStark

**Comment by OskarStark** [comment]
Please merge @greg0ire

**Comment by greg0ire** [comment]
Thanks @enekochan @OskarStark

File: Twig/Extension/FormatterMediaExtension.php [hunk 1]
```diff
@@ -76,14 +76,6 @@ public function getTwigExtension()
     }
 
     /**
-     * {@inheritdoc}
-     */
-    public function getName()
```
Comment (core23) [inline Twig/Extension/FormatterMediaExtension.php]: This won't work with twig 1.x @greg0ire

Comment (enekochan) [inline Twig/Extension/FormatterMediaExtension.php]: It should work on Twig >= 1.26 as it was marked as deprecated in that version. This PR has added Twig in `composer.json` like this  "twig/twig": "^1.28 || ^2.0" so there should be no problem IMO.

Comment (core23) [inline Twig/Extension/FormatterMediaExtension.php]: The method is deprecated, but `getName` still exists in the interface https://github.com/twigphp/Twig/blob/v1.28.0/lib/Twig/ExtensionInterface.php#L86

Comment (enekochan) [inline Twig/Extension/FormatterMediaExtension.php]: But those Twig extensions in SonataMediaBundle don't implement `ExtensionInterface`, but they extend `Twig_Extension` as recommended:

```php
class MediaExtension extends \Twig_Extension implements \Twig_Extension_InitRuntimeInterface
{
    ...
}
```

I tested this with Twig 1.31.0 and didn't have any problems (AFAIK). May be I'm not seeing the big picture.

Comment (core23) [inline Twig/Extension/FormatterMediaExtension.php]: The problem is the  `FormatterMediaExtension` class which extends `BaseProxyExtension`

Comment (enekochan) [inline Twig/Extension/FormatterMediaExtension.php]: So what would be a possible solution? Change `BaseProxyExtension` in SonataFormatterBundle not to implement `Twig_ExtensionInterface`?

[http://packanalyst.com/class?q=Sonata%5CFormatterBundle%5CExtension%5CBaseProxyExtension](http://packanalyst.com/class?q=Sonata%5CFormatterBundle%5CExtension%5CBaseProxyExtension)

Comment (core23) [inline Twig/Extension/FormatterMediaExtension.php]: That class should extend `\Twig_Extension` to solve the problem.

Comment (greg0ire) [inline Twig/Extension/FormatterMediaExtension.php]: This has been released and breaks things , it seems. Can you follow up on this? See #1215

**Pull request merged**